File: javascript/what_century(6).js
function whatCentury(year)
{
  var first_2_characters = year.toString().substring(0,2);
  var first_2_numbers = parseInt(first_2_characters);
  var century = first_2_numbers + 1;
  if(century.toString().slice(-1).match(/[0,4-9]/) || century.toString().charAt(0) == '1'){
  	return century.toString() + "th"
  }
  else if(century.toString().slice(-1).match(/[1]/)){
  	return century.toString() + "st"
  }
  else if(century.toString().slice(-1).match(/[2]/)){
  	return century.toString() + "nd"
  }
  else{
  	return century.toString() + "rd"
  }
}

debug(whatCentury(1999));
debug(whatCentury(2011));
debug(whatCentury(2154));
debug(whatCentury(2259));
debug(whatCentury(1124));
debug(whatCentury(1234));
debug(whatCentury(2000));

File: ruby/what_century(6).rb
def what_century(year)
  first_2_numbers = year.to_s[0,2]
  century = first_2_numbers.to_i + 1
  if century.to_s[1].match(/[0,4-9]/) || century.to_s[0] == '1'
    century.to_s + "th"
  elsif century.to_s[1].match(/[1]/)
    century.to_s + "st"
  elsif century.to_s[1].match(/[2]/) 
    century.to_s + "nd"
  else
    century.to_s + "rd"
  end
end

print what_century(1959)
print what_century(2011)
print what_century(2154)
print what_century(2259)
print what_century(1124)

# 0, 4, 5, 6, 7, 8, 9 th
# 1 st
# 2 nd
# 3 rd
